contrib/jfr_events/src/main/java/io/opentelemetry/contrib/jfr/JfrContextStorageListenerNew.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.jfr;
+
+import io.grpc.Context;
+import io.grpc.override.ContextStorageListener;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.unsafe.ContextUtils;
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class JfrContextStorageListenerNew implements ContextStorageListener.Provider {
```

Comment:
javadoc, and what's with this name? What makes it "new"?

Reply:
Did some updates to Bogdan's code handling context to make implementation cleaner. Deleted the old one and only keeping the new one (and renamed)